disperser/batcher/batch_confirmer.go

```diff
@@ -0,0 +1,531 @@
+package batcher
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"math/big"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/common"
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/Layr-Labs/eigenda/disperser"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/hashicorp/go-multierror"
+)
+
+type BatchConfirmerConfig struct {
+	PullInterval                 time.Duration
+	DispersalTimeout             time.Duration
+	DispersalStatusCheckInterval time.Duration
+	AttestationTimeout           time.Duration
+	SRSOrder                     int
+	NumConnections               int
+	MaxNumRetriesPerBlob         uint
+}
+
+type BatchConfirmer struct {
+	BatchConfirmerConfig
+
+	BlobStore      disperser.BlobStore
+	MinibatchStore MinibatchStore
+	Dispatcher     disperser.Dispatcher
+	EncoderClient  disperser.EncoderClient
+
+	ChainState         core.IndexedChainState
+	EncodingStreamer   *EncodingStreamer
+	Aggregator         core.SignatureAggregator
+	Transactor         core.Transactor
+	TransactionManager TxnManager
+	Minibatcher        *Minibatcher
+
+	ethClient common.EthClient
+	logger    logging.Logger
+}
+
+func NewBatchConfirmer(
+	config BatchConfirmerConfig,
+	blobStore disperser.BlobStore,
+	minibatchStore MinibatchStore,
+	dispatcher disperser.Dispatcher,
+	chainState core.IndexedChainState,
+	assignmentCoordinator core.AssignmentCoordinator,
+	encodingStreamer *EncodingStreamer,
+	aggregator core.SignatureAggregator,
+	ethClient common.EthClient,
+	transactor core.Transactor,
+	txnManager TxnManager,
+	minibatcher *Minibatcher,
+	logger logging.Logger,
+) (*BatchConfirmer, error) {
+	return &BatchConfirmer{
+		BatchConfirmerConfig: config,
+
+		BlobStore:        blobStore,
+		MinibatchStore:   minibatchStore,
+		Dispatcher:       dispatcher,
+		EncodingStreamer: encodingStreamer,
+
+		ChainState:         chainState,
+		Aggregator:         aggregator,
+		Transactor:         transactor,
+		TransactionManager: txnManager,
+		Minibatcher:        minibatcher,
+
+		ethClient: ethClient,
+		logger:    logger.With("component", "BatchConfirmer"),
+	}, nil
+}
+
+func (b *BatchConfirmer) Start(ctx context.Context) error {
+	err := b.ChainState.Start(ctx)
+	if err != nil {
+		return err
+	}
+	// Wait for few seconds for indexer to index blockchain
+	// This won't be needed when we switch to using Graph node
+	time.Sleep(indexerWarmupDelay)
```

Comment:
Is this still used?

Reply:
It _could_ still be used if we wanted to (i.e. testing)